### ISSUE TITLE
seafdav partial download fails: ''SeafBlockManager' object has no attribute 'obj_store''

### ISSUE THREAD

**Issue by why-1998** [haiwen/seafobj#52]
HTTP 500: Source exception: ''SeafBlockManager' object has no attribute 'obj_store'' is the error when downloading partial content via Seafile WebDAV and the byte range does not start at 0.
A place in the source code may be "blocks.py" where stat_block(self, repo_id, version, obj_id) only treats "obj_store" in case of not objstore_factory.enable_storage_classes but not "obj_stores"?

**Comment by why-1998** [comment]
No one to have a look for the pull request #53 to check if this would pass test (or may be it is wrong)?

test script to reproduce the misbehavior:

```
#!/bin/bash
BASIC_AUTH_USER="user@seafile.domain.tld"
BASIC_AUTH_PASSWORD="seafdavpassword"
AUTH=$(echo -ne "$BASIC_AUTH_USER:$BASIC_AUTH_PASSWORD" | base64 --wrap 0)
curl -v --header "Authorization: Basic $AUTH" --header "Range: bytes=0-1023" --header "Referer: https://seafile.domain.tld/seafdav/" "https://seafile.domain.tld/seafdav/mylibrary/thefiletodownloadpartial.dat" --output thefiletodownloadpartial.dat_part_OK
curl -v --header "Authorization: Basic $AUTH" --header "Range: bytes=1-1023" --header "Referer: https://seafile.domain.tld/seafdav/" "https://seafile.domain.tld/seafdav/mylibrary/thefiletodownloadpartial.dat" --output thefiletodownloadpartial.dat_part_NOTWORKING
```

The first curl will get the correct bytes from the server, the second one catches an exception and results in the HTTP 500 error saying:
**Source exception: ''SeafBlockManager' object has no attribute 'obj_store''**

Please have a look at the code of #53. I really looks like the branching on single/multiple backends needs to be checked in blocks.py at stat_block.

**Comment by why-1998** [comment]
Has been fixed with #56.
